lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -18,10 +18,24 @@ export default class DocumentComposer {
   /**
    * Transforms the given DID state into a DID Document.
    */
-  public static transformToExternalDocument (didState: DidState, did: string, published: boolean): any {
+  public static transformToExternalDocument (didState: DidState, did: string, published: boolean, isShortForm: boolean): any {
     // If the DID is deactivated.
+    // Return required metadata and a document with only context and id if so
     if (didState.nextRecoveryCommitmentHash === undefined) {
-      return { status: 'deactivated' };
+      const didDocument = {
```

Comment:
Consider move this block into its own method perhaps `generateDeactivatedResolutionObject(...)` since this method is becoming large.